src/modules/gimbal/gimbal.cpp

```diff
@@ -211,7 +211,10 @@ static int gimbal_thread_main(int argc, char *argv[])
 			// Reset control as no one is active anymore, or yet.
 			thread_data.control_data.sysid_primary_control = 0;
 			thread_data.control_data.compid_primary_control = 0;
-			thread_data.control_data.device_compid = 0;
+			// If the output is set to AUX we still want the input to be able to control the gimbal
+			// via mavlink, so we set the device_compid to 1. Basically in this configuration both
+			// gimbal manager and gimbal device have the same component ID as the autopilot (=1).
```

Comment:
I think the solution is good but the last line of the comment is not correct as such.

According to the protocol, the device_compid should not be "the same as the autopilot" but it should be any number from 1 to 6 to signal a non-MAVLink gimbal (which this is).

See: https://mavlink.io/en/messages/common.html#GIMBAL_MANAGER_INFORMATION

> Gimbal device ID that this gimbal manager is responsible for. Component ID of gimbal device (or 1-6 for non-MAVLink gimbal).

Reply:
And thanks for the PR @DanielePettenuzzo, appreciated!

Reply:
@julianoes Thank you for the review! What you are saying makes sense. I changed the comment. When we are good to merge we can then squash the last commit.